src/media/js/carrier.js

```diff
@@ -0,0 +1,67 @@
+define('carrier',
+    ['consumer_info', 'core/defer', 'core/requests', 'core/storage',
+     'core/urls', 'core/user', 'underscore', 'user_helpers'],
+    function(consumerInfo, defer, requests, storage,
+             urls, user, _, userHelpers) {
+    'use strict';
+
+    function notInstalled(app) {
+        return !user.has_installed(app.id);
+    }
+
+    function wait(time) {
+        var def = defer.Deferred();
+        setTimeout(function() {
+            def.resolve();
+        }, time);
+        return def;
+    }
+
+    var carrier = userHelpers.carrier();
+    var region = userHelpers.region();
+    var STORAGE_KEY = 'late-customization-complete';
+    var apiResponse;
+
+    if (carrier && region && !isLateCustomizationCompleted()) {
+        apiResponse = requests.get(urls.api.url('late-customization', [], {
+            carrier: carrier,
+            region: region,
+        }));
+    } else {
+        apiResponse = defer.Deferred().reject();
+    }
+
+    var allApps = apiResponse.then(function(response) {
+        return response.objects;
+    });
+
+    function hasInstallableApps() {
+        // We need to wait for a bit each time this is called because the
+        // install-success event fires before the installed apps list is
+        // updated.
+        return defer
+            .when(allApps, consumerInfo.promise, wait(20))
+            .then(function(apps) {
+                var installable = _.some(apps, notInstalled);
+                if (!installable) {
+                    lateCustomizationCompleted();
+                }
+                return installable;
+            });
+    }
+
+    function lateCustomizationCompleted() {
```

Comment:
I think I made it past tense so that it's clear that calling it won't return a boolean. Although after reading it now that actually isn't any more clear. Late customisation is far more verbose than I would like.